### ISSUE TITLE
Why does the evaluation metric change to 0 or nan during validation？How to solve it?

### ISSUE THREAD

**Issue by 0xf21** [XiandaGuo/OpenStereo#226]
**Why does the evaluation metric change to 0 or nan during validation after training on the InStereo2K dataset for over a hundred epochs? How to solve it?**

This is the **training log**:
```shell
2025-04-15 17:56:34,439   INFO  Training Epoch:141/500 Iter:   6/84 Loss:0.336132(0.453223) LR:1.9796e-03 DataTime:0.05 InferTime:15.03ms Time cost: 1:13:24/3:06:44
2025-04-15 17:56:37,654   INFO  Training Epoch:141/500 Iter:  16/84 Loss:0.379276(0.460977) LR:1.9789e-03 DataTime:0.05 InferTime:15.02ms Time cost: 1:13:27/3:06:39
2025-04-15 17:56:40,871   INFO  Training Epoch:141/500 Iter:  26/84 Loss:0.440607(0.456168) LR:1.9782e-03 DataTime:0.05 InferTime:15.01ms Time cost: 1:13:30/3:06:34
2025-04-15 17:56:44,065   INFO  Training Epoch:141/500 Iter:  36/84 Loss:0.609625(0.460095) LR:1.9775e-03 DataTime:0.05 InferTime:15.03ms Time cost: 1:13:33/3:06:29
2025-04-15 17:56:47,228   INFO  Training Epoch:141/500 Iter:  46/84 Loss:0.533846(0.469345) LR:1.9768e-03 DataTime:0.05 InferTime:15.01ms Time cost: 1:13:37/3:06:24
2025-04-15 17:56:50,437   INFO  Training Epoch:141/500 Iter:  56/84 Loss:0.586915(0.474193) LR:1.9761e-03 DataTime:0.05 InferTime:15.01ms Time cost: 1:13:40/3:06:19
2025-04-15 17:56:53,614   INFO  Training Epoch:141/500 Iter:  66/84 Loss:0.329534(0.468878) LR:1.9754e-03 DataTime:0.05 InferTime:15.04ms Time cost: 1:13:43/3:06:14
2025-04-15 17:56:56,813   INFO  Training Epoch:141/500 Iter:  76/84 Loss:0.417967(0.482099) LR:1.9747e-03 DataTime:0.04 InferTime:15.01ms Time cost: 1:13:46/3:06:09
2025-04-15 17:57:00,520   INFO  Evaluating Epoch:141 Iter:   0/5 InferTime: 13.15ms
2025-04-15 17:57:01,533   INFO  Epoch 141 metrics: {'d1_all': tensor(1.6523), 'epe': tensor(0.8125), 'thres_1': tensor(14.3116), 'thres_2': tensor(6.5153), 'thres_3': tensor(4.2214)}
epochs:  28%|████████████████████▏                                                  | 142/500 [1:13:51<3:05:39, 31.12s/it]
2025-04-15 17:57:04,273   INFO  Training Epoch:142/500 Iter:   2/84 Loss:nan(nan) LR:1.9741e-03 DataTime:0.06 InferTime:26.55ms Time cost: 1:13:54/3:06:14
OpenStereo/./stereo/utils/common_utils.py:198: RuntimeWarning: invalid value encountered in cast
  pred_tmp = cm(pred_tmp.astype('uint8'))
OpenStereo/./stereo/utils/common_utils.py:199: RuntimeWarning: invalid value encountered in cast
  error_map_tmp = cm(error_map_tmp.astype('uint8'))
2025-04-15 17:57:07,509   INFO  Training Epoch:142/500 Iter:  12/84 Loss:0.345061(nan) LR:1.9734e-03 DataTime:0.04 InferTime:15.02ms Time cost: 1:13:57/3:06:10
2025-04-15 17:57:10,718   INFO  Training Epoch:142/500 Iter:  22/84 Loss:0.619040(nan) LR:1.9727e-03 DataTime:0.05 InferTime:15.57ms Time cost: 1:14:00/3:06:05
2025-04-15 17:57:13,957   INFO  Training Epoch:142/500 Iter:  32/84 Loss:0.422575(nan) LR:1.9720e-03 DataTime:0.04 InferTime:15.02ms Time cost: 1:14:03/3:06:00
2025-04-15 17:57:17,124   INFO  Training Epoch:142/500 Iter:  42/84 Loss:0.369364(nan) LR:1.9713e-03 DataTime:0.05 InferTime:24.99ms Time cost: 1:14:06/3:05:55
2025-04-15 17:57:20,288   INFO  Training Epoch:142/500 Iter:  52/84 Loss:0.378441(nan) LR:1.9706e-03 DataTime:0.04 InferTime:15.02ms Time cost: 1:14:10/3:05:49
2025-04-15 17:57:23,453   INFO  Training Epoch:142/500 Iter:  62/84 Loss:0.489422(nan) LR:1.9699e-03 DataTime:0.05 InferTime:15.03ms Time cost: 1:14:13/3:05:44
2025-04-15 17:57:26,628   INFO  Training Epoch:142/500 Iter:  72/84 Loss:0.516815(nan) LR:1.9692e-03 DataTime:0.04 InferTime:15.01ms Time cost: 1:14:16/3:05:39
2025-04-15 17:57:29,773   INFO  Training Epoch:142/500 Iter:  82/84 Loss:0.550055(nan) LR:1.9685e-03 DataTime:0.04 InferTime:15.00ms Time cost: 1:14:19/3:05:34
2025-04-15 17:57:31,670   INFO  Evaluating Epoch:142 Iter:   0/5 InferTime: 13.06ms
2025-04-15 17:57:32,560   INFO  Epoch 142 metrics: {'d1_all': tensor(0.), 'epe': tensor(nan), 'thres_1': tensor(0.), 'thres_2': tensor(0.), 'thres_3': tensor(0.)}
epochs:  29%|████████████████████▎                                                  | 143/500 [1:14:22<3:04:58, 31.09s/it]
2025-04-15 17:57:37,209   INFO  Training Epoch:143/500 Iter:   8/84 Loss:0.406857(0.454781) LR:1.9678e-03 DataTime:0.07 InferTime:25.37ms Time cost: 1:14:27/3:05:40
2025-04-15 17:57:40,407   INFO  Training Epoch:143/500 Iter:  18/84 Loss:0.374003(0.466493) LR:1.9671e-03 DataTime:0.05 InferTime:15.07ms Time cost: 1:14:30/3:05:35
2025-04-15 17:57:43,592   INFO  Training Epoch:143/500 Iter:  28/84 Loss:0.306374(0.472204) LR:1.9664e-03 DataTime:0.05 InferTime:15.02ms Time cost: 1:14:33/3:05:30
2025-04-15 17:57:46,754   INFO  Training Epoch:143/500 Iter:  38/84 Loss:0.451911(0.454139) LR:1.9657e-03 DataTime:0.04 InferTime:15.02ms Time cost: 1:14:36/3:05:25
2025-04-15 17:57:49,994   INFO  Training Epoch:143/500 Iter:  48/84 Loss:0.459635(0.444531) LR:1.9650e-03 DataTime:0.04 InferTime:15.02ms Time cost: 1:14:39/3:05:20
2025-04-15 17:57:53,177   INFO  Training Epoch:143/500 Iter:  58/84 Loss:0.436127(0.449128) LR:1.9643e-03 DataTime:0.05 InferTime:15.03ms Time cost: 1:14:42/3:05:15
2025-04-15 17:57:56,338   INFO  Training Epoch:143/500 Iter:  68/84 Loss:0.418680(0.451135) LR:1.9636e-03 DataTime:0.04 InferTime:15.03ms Time cost: 1:14:46/3:05:10
2025-04-15 17:57:59,477   INFO  Training Epoch:143/500 Iter:  78/84 Loss:0.455974(0.454623) LR:1.9629e-03 DataTime:0.04 InferTime:15.02ms Time cost: 1:14:49/3:05:04
2025-04-15 17:58:02,558   INFO  Evaluating Epoch:143 Iter:   0/5 InferTime: 13.28ms
2025-04-15 17:58:03,451   INFO  Epoch 143 metrics: {'d1_all': tensor(0.), 'epe': tensor(nan), 'thres_1': tensor(0.), 'thres_2': tensor(0.), 'thres_3': tensor(0.)}
``` 

This is the **yaml**：
```shell
DATA_CONFIG:
    DATA_INFOS:
        -   DATASET: InStereo2KDataset
            DATA_SPLIT: {
                TRAINING: ./data/InStereo2K/InStereo2K_train.txt,
                EVALUATING: ./data/InStereo2K/InStereo2K_test.txt,
                TESTING: ./data/InStereo2K/InStereo2K_test.txt
            }
            DATA_PATH: ./data/datasets/stereo/InStereo2K
            RETURN_RIGHT_DISP: false

    DATA_TRANSFORM:
        TRAINING:
            - { NAME: RandomCrop, SIZE: [ 480, 640 ], Y_JITTER: false }
            - { NAME: TransposeImage }
            - { NAME: ToTensor }
            - { NAME: NormalizeImage, MEAN: [ 0.485, 0.456, 0.406 ], STD: [ 0.229, 0.224, 0.225 ] }
        EVALUATING:
            - { NAME: RightTopPad, SIZE: [ 480, 640 ] }
            - { NAME: TransposeImage }
            - { NAME: ToTensor }
            - { NAME: NormalizeImage, MEAN: [ 0.485, 0.456, 0.406 ], STD: [ 0.229, 0.224, 0.225 ] }

MODEL:
    NAME: LightStereo
    MAX_DISP: 192
    EXPANSE_RATIO: 4
    AGGREGATION_BLOCKS: [1, 2, 4]
    LEFT_ATT: true
    FIND_UNUSED_PARAMETERS: false
    CKPT: -1
    PRETRAINED_MODEL: './weights/LightStereo-S-SceneFlow.ckpt'

OPTIMIZATION:
    FREEZE_BN: false
    SYNC_BN: true
    AMP: true
    BATCH_SIZE_PER_GPU: 24
    NUM_EPOCHS: 500

    OPTIMIZER:
        NAME: AdamW
        LR: &lr 0.0024
        WEIGHT_DECAY: 1.0e-05
        EPS: 1.0e-08

    SCHEDULER:
        NAME: OneCycleLR
        MAX_LR: *lr
        PCT_START: 0.01
        ON_EPOCH: False

    CLIP_GRAD:
        TYPE: value
        CLIP_VALUE: 0.1

EVALUATOR:
    BATCH_SIZE_PER_GPU: 12
    MAX_DISP: 192
    METRIC:
        - d1_all
        - epe
        - thres_1
        - thres_2
        - thres_3

TRAINER:
    EVAL_INTERVAL: 1
    CKPT_SAVE_INTERVAL: 1
    MAX_CKPT_SAVE_NUM: 30
    LOGGER_ITER_INTERVAL: 10
    TRAIN_VISUALIZATION: True
    EVAL_VISUALIZATION: True
```

**Comment by XiandaGuo** [comment]
Thank you for your attention to our work. You can try reducing the learning rate (lr).
